apps/meteor/tests/e2e/omnichannel/omnichannel-take-chat.spec.ts

```diff
@@ -25,50 +27,66 @@ test.describe('omnichannel-takeChat', () => {
 
 	test.afterAll(async ({ api }) => {
 		await Promise.all([
+			await api.delete('/livechat/users/agent/user1').then((res) => expect(res.status()).toBe(200)),
 			await api.post('/settings/Livechat_Routing_Method', { value: 'Auto_Selection' }).then((res) => expect(res.status()).toBe(200)),
 			await api.post('/settings/Livechat_enabled_when_agent_idle', { value: false }).then((res) => expect(res.status()).toBe(200)),
-			await api.delete('/livechat/users/agent/user1').then((res) => expect(res.status()).toBe(200)),
 		]);
 
 		await agent.page.close();
 	});
 
-	test.beforeEach(async ({ page, api }) => {
-		// make "user-1" online
+	test.beforeEach('start a new livechat chat', async ({ page, api },) => {
 		await agent.poHomeChannel.sidenav.switchStatus('online');
 
-		// start a new chat for each test
 		newVisitor = {
 			name: `${faker.person.firstName()} ${faker.string.uuid()}`,
 			email: faker.internet.email(),
 		};
+
 		poLiveChat = new OmnichannelLiveChat(page, api);
+
 		await page.goto('/livechat');
-		await poLiveChat.openLiveChat();
-		await poLiveChat.sendMessage(newVisitor, false);
-		await poLiveChat.onlineAgentMessage.type('this_a_test_message_from_user');
-		await poLiveChat.btnSendMessageToOnlineAgent.click();
 	});
 
-	test('expect "user1" to be able to take the chat from the queue', async () => {
+	test('should user take the chat', async () => {
+		await poLiveChat.openChatAndSendMessage(newVisitor, 'this_a_test_message_from_user');
+
 		await agent.poHomeChannel.sidenav.openQueuedOmnichannelChat(newVisitor.name);
 		await expect(agent.poHomeChannel.content.btnTakeChat).toBeVisible();
 		await agent.poHomeChannel.content.btnTakeChat.click();
 
 		await agent.poHomeChannel.sidenav.openChat(newVisitor.name);
 		await expect(agent.poHomeChannel.content.btnTakeChat).not.toBeVisible();
 		await expect(agent.poHomeChannel.content.inputMessage).toBeVisible();
+
+		await poLiveChat.closeChat();
+	});
+
+	test('should user not receive a new chat when offline status and livechat disabled when agent idle', async ({ api }) => {
+		await api.post('/settings/Livechat_enabled_when_agent_idle', { value: false }).then((res) => expect(res.status()).toBe(200));
+		await agent.poHomeChannel.sidenav.switchStatus('offline');
+
+		await poLiveChat.openChatAndSendMessage(newVisitor, 'this_a_test_message_from_user');
+
+		await sleep(2000);
```

Comment:
Would not this be a problem? 🤔

Reply:
I'm trying to avoid an always falsy test... Since the message from livechat could take some time to be received by RC instance 🤔

Reply:
playwright already waits for the elements, you can modify the timeout time if you believe it will make a difference, but honestly, waiting longer should not be the solution that will save your test